homeassistant/components/dexcom/sensor.py

```diff
@@ -0,0 +1,143 @@
+"""Support for Dexcom sensors."""
+from homeassistant.const import CONF_UNIT_OF_MEASUREMENT, CONF_USERNAME, STATE_UNKNOWN
+from homeassistant.helpers.entity import Entity
+
+from .const import DOMAIN, GLUCOSE_TREND_ICON, GLUCOSE_VALUE_ICON, MG_DL
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up the Dexcom sensors."""
+    coordinator = hass.data[DOMAIN][config_entry.entry_id]
+    username = config_entry.data[CONF_USERNAME]
+    unit_of_measurement = config_entry.data[CONF_UNIT_OF_MEASUREMENT]
+    sensors = []
+    sensors.append(DexcomGlucoseTrendSensor(coordinator, username))
+    sensors.append(DexcomGlucoseValueSensor(coordinator, username, unit_of_measurement))
+    async_add_entities(sensors, False)
+
+
+class DexcomGlucoseValueSensor(Entity):
+    """Representation of a Dexcom glucose value sensor."""
+
+    def __init__(self, coordinator, username, unit_of_measurement):
+        """Initialize the sensor."""
+        self._state = None
+        self._unit_of_measurement = unit_of_measurement
+        self._attribute_unit_of_measurement = (
+            "mg_dl" if unit_of_measurement == MG_DL else "mmol_l"
+        )
+        self._coordinator = coordinator
+        self._name = f"{DOMAIN}_{username}_glucose_value"
+        self._unique_id = f"{username}-value"
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return self._name
+
+    @property
+    def icon(self):
+        """Return the icon for the frontend."""
+        return GLUCOSE_VALUE_ICON
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement of the device."""
+        return self._unit_of_measurement
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return (
+            getattr(self._coordinator.data, self._attribute_unit_of_measurement)
+            if self._coordinator.data
+            else STATE_UNKNOWN
```

Comment:
Please use `None` to represent unknown state.

Please don't use a ternary operator condition if it spans more than one line.